newIDE/app/src/GameDashboard/GameAnalyticsEvaluator.js

```diff
@@ -0,0 +1,640 @@
+              ? d0PlayersBelow900s - d0PlayersBelow600s
+              : 0,
+          from900sToInfinityPlayersCount:
+            d0PlayersBelow900s !== undefined
+              ? d0Players - d0PlayersBelow900s
+              : 0,
+
+          over0sPlayersPercent: 100,
+          over60sPlayersPercent:
+            d0PlayersBelow60s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow60s)) / d0Players
+              : 0,
+          over180sPlayersPercent:
+            d0PlayersBelow180s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow180s)) / d0Players
+              : 0,
+          over300sPlayersPercent:
+            d0PlayersBelow300s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow300s)) / d0Players
+              : 0,
+          over600sPlayersPercent:
+            d0PlayersBelow600s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow600s)) / d0Players
+              : 0,
+          over900sPlayersPercent:
+            d0PlayersBelow900s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow900s)) / d0Players
+              : 0,
+          below60sPlayersPercent:
+            d0PlayersBelow60s !== undefined && d0Players > 0
+              ? (100 * d0PlayersBelow60s) / d0Players
+              : 0,
+          from60sTo180sPlayersPercent:
+            d0PlayersBelow180s !== undefined &&
+            d0PlayersBelow60s !== undefined &&
+            d0Players > 0
+              ? (100 * (d0PlayersBelow180s - d0PlayersBelow60s)) / d0Players
+              : 0,
+          from180sTo300sPlayersPercent:
+            d0PlayersBelow300s !== undefined &&
+            d0PlayersBelow180s !== undefined &&
+            d0Players > 0
+              ? (100 * (d0PlayersBelow300s - d0PlayersBelow180s)) / d0Players
+              : 0,
+          from300sTo600sPlayersPercent:
+            d0PlayersBelow600s !== undefined &&
+            d0PlayersBelow300s !== undefined &&
+            d0Players > 0
+              ? (100 * (d0PlayersBelow600s - d0PlayersBelow300s)) / d0Players
+              : 0,
+          from600sTo900sPlayersPercent:
+            d0PlayersBelow900s !== undefined &&
+            d0PlayersBelow600s !== undefined &&
+            d0Players > 0
+              ? (100 * (d0PlayersBelow900s - d0PlayersBelow600s)) / d0Players
+              : 0,
+          from900sToInfinityPlayersPercent:
+            d0PlayersBelow900s !== undefined && d0Players > 0
+              ? (100 * (d0Players - d0PlayersBelow900s)) / d0Players
+              : 0,
+        };
+      })
+      .sort((a, b) => a.timestamp - b.timestamp),
+    overPlayedDuration: [
+      { duration: 0, playersCount: onlyFullyDefinedPlayersSum },
+    ].concat(
+      // $FlowFixMe durationIndex can only be a number.
+      Object.values(durationIndexes).map((durationIndex: number) => ({
```

Comment:
it's weird that it's complaining as you've typed the durationIndexes correctly. Is it saying that it can be null or undefined ? Or something else?

Reply:
It seems that `values()` is always something like `any`.
```
Cannot call `Object.values(...).map` with function bound to `callbackfn` because number [1] is incompatible with
mixed [2] in the first argument. [incompatible-call]

   src/GameDashboard/GameAnalyticsEvaluator.js:603:58
   603|       Object.values(durationIndexes).map((durationIndex: number) => ({                                                                 ^^^^^^ [1]   

References:
   C:\Users\davyh\AppData\Local\Temp\flow\flowlib_1c1169e0\core.js:80:50      
    80|     static values(object: $NotNullOrVoid): Array<mixed>;
                                                         ^^^^^ [2]
```